### PR TITLE
Transactor provider

### PR DESCRIPTION
**Pull request by autholykos**
part of #470

### CODE REVIEW

**Comment by autholykos** [comment]
fixes #472

File: pkg/core/data/transactions/fields.go [hunk 1]
```diff
@@ -288,6 +288,11 @@ type SecretKey struct {
 	B *Scalar `json:"b"`
 }
 
+// IsEmpty is used when passing the SecretKey by value
+func (s SecretKey) IsEmpty() bool {
+	return s.A == nil && s.B == nil
+}
+
 // MSecretKey copy the USecretKey from rusk to transactions datastruct
```
Comment (julesdesmit) [inline pkg/core/data/transactions/fields.go:296]: This comment is the same as the one for `USecretKey`, so it's a bit hard to know which one is going in which direction unless you look at the logic

File: pkg/core/data/transactions/fields.go [hunk 2]
```diff
@@ -288,6 +288,11 @@ type SecretKey struct {
...
 func MSecretKey(r *rusk.SecretKey, n *SecretKey) {
 	r.A = new(rusk.Scalar)
```
Comment (julesdesmit) [inline pkg/core/data/transactions/fields.go:298]: Should we have the `IsEmpty` check here as well, like in `MPublicKey`?

File: pkg/core/mempool/mempool.go [hunk 3]
```diff
@@ -65,37 +63,43 @@ type Mempool struct {
 	latestBlockTimestamp int64
 
 	eventBus *eventbus.EventBus
-	db       database.DB
+	// TODO: do we still need a DB in the mempool?
```
Comment (julesdesmit) [inline pkg/core/mempool/mempool.go:66]: Nope, as discussed, Rusk will check the DB for double-spends when verifying a mempool tx.

File: pkg/core/data/transactions/fields.go (hunk 1 above)
Comment (autholykos) [reply pkg/core/data/transactions/fields.go:296]: good point, will fix

File: pkg/core/data/transactions/fields.go (hunk 2 above)
Comment (autholykos) [reply pkg/core/data/transactions/fields.go:298]: In theory sounds good. However, the `SecretKey` is normally required. I will fix it but we might want to add some verification by the caller

File: pkg/core/mempool/mempool.go (hunk 3 above)
Comment (autholykos) [reply pkg/core/mempool/mempool.go:66]: Removed the comment

**Comment by autholykos** [comment]
Fixes #475

**Comment by autholykos** [comment]
Solves #479

**Comment by autholykos** [comment]
Fixes #456

**Pull request merged**